components/TableOfContent.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import { TableContentModel } from '@models/TableOfConent.model';
+import TableOfContentListItem from '@components/TableOfContentListItem';
+import { v4 } from 'uuid';
+
+const TableOfContent = ({
+    titles,
+    active,
+    setActive
+}: {
+    titles: TableContentModel[];
+    active: string;
+    setActive: Dispatch<SetStateAction<null | string>>;
+}): JSX.Element => {
+    if (!titles?.length) {
```

Comment:
Instead of optional chaining and early return you can default `titles` to an empty array.

Reply:
I set an empty array by default, but I left this conditional: `!titles.length` because if this component has an empty array it will return an empty <ul> element.

Reply:
@rjvelazco Yeah that's not a problem, check for `titles.length in the `return` and return `null` of `titles` is falsy.

`return {titles.length ? <li>TableOfContentListItem</li> : null}`